src/components/LoginPage/index.js

```diff
@@ -0,0 +1,94 @@
+import { Section } from "@commons-ui/core";
+import { Button, Typography, Grid } from "@material-ui/core";
+import { useSession, signIn } from "next-auth/react";
+import Image from "next/image";
+import Router from "next/router";
+import PropTypes from "prop-types";
+import React, { useEffect, useState } from "react";
+
+import useStyles from "./useStyles";
+
+import googleIcon from "@/promisetracker/assets/google-icon.svg";
+import LoginForm from "@/promisetracker/components/LoginForm";
+import getErrorDescription from "@/promisetracker/utils/auth/error-list";
+
+function Login({
+  providers: providersProp,
+  loginLabel,
+  description,
+  ...props
+}) {
+  const classes = useStyles(props);
+  const { data: session } = useSession();
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  useEffect(() => {
+    setErrorMessage(getErrorDescription(Router.query?.error));
+  }, []);
+
+  useEffect(() => {
+    if (session) {
+      Router.push("/act-now");
+    }
+  }, [session]);
+
+  const providers = Object.values(providersProp ?? {});
```

Comment:
```suggestion
  const providers = providersProp ? Object.values(providersProp) : undefined;
```
Why waste {} (and the empty [] returned from Object.values)?